H-2367: Mention Rust exceptions in `Fractal Trees` HDEV post (#4153)
hashintel/hash: apps/hashdotdev/src/_pages/blog/0016_file-structuring.mdx

- In [Þ → libs/blockprotocol](https://github.com/blockprotocol/blockprotocol/tree/main/libs/blockprotocol)
- In [# → @hashintel/design-system](https://github.com/hashintel/hash/tree/main/libs/@hashintel/design-system)

## Co-existing with Rust in a monorepo

Our fractal tree approach applies only to the JavaScript portion of our monorepo. Not all rules hold in Rust. Key differences in our Rust code include:

- **P4.** resources of a sub-sub-tree are inaccessible
- **Rule 2:** no `index` files are used (we tried this, but found it more distracting and a worse developer experience than simply just using `mod.rs`)
- **Rule 5:** imports within sub-trees do not use relative paths (given increased refactoring costs, and by default imports are absolute in Rust with the exception of reexporting sub-module exports)
- **Rule 6:** we don't have a shared folder in Rust, and we want to avoid having one
- **Rule 7:** multiple entry points are disallowed (our entry point is `fn main()` - always)

> - **Rule 8:** unit tests typically are located at the bottom of the file inside of a `test` module, this is a Rust convention

Author: ```suggestion
- **Rule 8:** unit tests are typically colocated at the _bottom_ of files inside of a `test` module, per Rust convention
```
